packages/runtime/container-runtime/src/summarizerTypes.ts

```diff
@@ -132,6 +133,43 @@ export type GenerateSummaryResult =
     | IUploadSummaryResult
     | ISubmitSummaryOpResult;
 
+export interface IBroadcastSummaryResult {
+    readonly summarizeOp: ISummaryOpMessage;
+    readonly broadcastDuration: number;
+}
+
+export interface IAckNackSummaryResult {
+    readonly summaryAckNackOp: ISummaryAckMessage | ISummaryNackMessage;
+    readonly ackNackDuration: number;
+}
+
+export type SummarizeResultPart<T> = {
+    success: true;
+    data: T;
+} | {
+    success: false;
+    data: T | undefined;
+    message: string;
+    error: any;
+};
+
+export interface ISummarizeResult {
+    /** Resolves when we generate, upload, and submit the summary */
+    readonly generateSummary: Promise<SummarizeResultPart<GenerateSummaryResult>>;
```

Comment:
GenerateSummaryResult does not include the summaryTree that was generated. For testing, we will want access to the summaryTree right.

Reply:
Oops! I thought the stats type had the tree too... good catch; will fix!

Reply:
Added the summaryTree and gcData.